mindsdb/integrations/libs/ml_exec_base.py

```diff
@@ -311,16 +311,27 @@ def learn(self, statement):
         if target not in training_data_df.columns:
             raise Exception(f'Prediction target "{target}" not found in training dataframe: {list(training_data_df.columns)}')
 
-        problem_definition = statement.using
-        if problem_definition is None:
-            problem_definition = {}
+        problem_definition = {'target': target}
 
-        problem_definition['target'] = target
+        for attr in ['using', 'horizon', 'window']:
+            if hasattr(statement, attr) and getattr(statement, attr) is not None:
+                problem_definition[attr] = getattr(statement, attr)
+
+        if hasattr(statement, 'order_by') and statement.order_by is not None:
```

Comment:
You can remove 'hasattr(statement, 'order_by')' and the similar because statement is CreatePredictor and already has these attributes